CONTRIBUTING.md

```diff
@@ -1,27 +1,167 @@
+own the copyright — so you retain the right to modify your code and use it in
+other projects.
 
-Static test use several code analysis tools present in the GoLang ecosystem
+The agreement can be found, and signed, here:
+https://ubuntu.com/legal/contributors
+
+If you have any questions, please reach out to us on our forum:
+https://forum.snapcraft.io/c/snapd/5
+
+## Contributor guidelines
+
+Contributors can help us by observing the following guidelines:
+
+- Commit messages should be well structured.
+- Commit emails should not include non-ASCII characters.
+- PR titles should start with a prefix denoting the scope affected by the changes
+(e.g. `<feature name>`,`<package name>`) followed by a semicolon. 
+- Try to open several smaller PRs, rather than one large PR.
+- Try not to mix potentially controversial and trivial changes together.
+  (Proposing trivial changes separately makes landing them easier and 
+  makes reviewing controversial changes simpler)
+- Try not to [force push][git-force] to a PR after it has received reviews. It is
+  acceptable to force push when a PR is ready to merge, however.
+- Try to write tests to cover the contributed changes (see below)
+
+## Pull requests and tests
+
+Before merging any changes into the snapd codebase, we need to verify that the
+proposed functionality and code quality does not degrade the functionality and
+quality requirement we've set for the project.
+
+For each PR, we run checks in three different groups: static, unit and spread.
+
+Static tests use several code analysis tools present in the GoLang ecosystem
 (go vet, go lint and go fmt) to make sure that the code always aligns with
 the standards. They also check the markdown format of documentation files.
+
 All the existing unit tests are also executed, and the coverage info is
-reported to coveralls. Regarding [spread](https://github.com/snapcore/spread),
-we use it to verify the integrity of the product exercising it as a whole,
-both from an end user standpoint (eg. all kind of interactions with the
-snap tool from the command line) and from a more systemic approach (testing
-upgrades, for instance).
-
-We do not set as a requirement the addition of spread and unit tests for a PR
-to be merged, but encourage the contributors to add them so that the expected
-behaviour is explained and verified through the tests and the review process
-can be made on the solid base of a working system after the addition of the
-changes. If any tests need to be added for a PR to be merged it will be denoted
+reported to coveralls.
+
+We use [spread](https://github.com/snapcore/spread) to verify the
+integrity of the product, exercising it as a whole, both from an end user
+standpoint (eg. all kinds of interactions with the snap tool from the command
+line) and from a more systemic approach (testing upgrades, for instance).
+
+We do not set the addition of spread and unit tests as a requirement for a PR
```

Comment:
I'm not sure this is true, we kind of do, we might add them on behalf of the contributor but ultimately it's very rare that we land code without tests

Reply:
I suppose the point I'm trying to make is about the "merge" verb in the sentence. It seems truer if the verb was "submitted" instead for example.

Reply:
Thanks. I've tweaked this to hopefully make it more explicit.